File: backend/src/main/java/com/offshore/platform/common/enums/ConflictType.java
package com.offshore.platform.common.enums;

/**
 * Sync conflict type — shared by backend, mobile, and PC frontend.
 *
 * Backend: sync_conflict.conflict_type
 */
public enum ConflictType {

    VERSION_CONFLICT("VERSION_CONFLICT", "版本冲突"),
    FIELD_CONFLICT("FIELD_CONFLICT", "字段冲突"),
    UPDATE_AFTER_DELETE("UPDATE_AFTER_DELETE", "删除后更新冲突"),
    DELETE_AFTER_UPDATE("DELETE_AFTER_UPDATE", "更新后删除冲突"),
    DUPLICATE_CREATE("DUPLICATE_CREATE", "重复创建冲突"),
    PERMISSION_CONFLICT("PERMISSION_CONFLICT", "权限冲突"),
    FILE_META_CONFLICT("FILE_META_CONFLICT", "文件元数据冲突"),
    ACCEPTANCE_LOCKED_CONFLICT("ACCEPTANCE_LOCKED_CONFLICT", "验收锁定冲突");

    private final String code;
    private final String message;

    ConflictType(String code, String message) {
        this.code = code;
        this.message = message;
    }

    public String getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public static ConflictType fromCode(String code) {
        for (ConflictType type : values()) {
            if (type.code.equals(code)) {
                return type;
            }
        }
        return VERSION_CONFLICT;
    }
}

File: mobile/app/src/main/java/com/offshore/platform/mobile/domain/enums/ConflictType.kt
package com.offshore.platform.mobile.domain.enums

import androidx.compose.ui.graphics.Color
import com.offshore.platform.mobile.ui.theme.*

/**
 * Sync conflict type — matches backend.
 *
 * Backend: sync_conflict.conflict_type
 * Database: database-docs/02_Status_Enum_Dictionary.md
 */
enum class ConflictType(val code: String, val displayName: String, val color: Color) {
    VERSION_CONFLICT("VERSION_CONFLICT", "版本冲突", StatusWarning),
    FIELD_CONFLICT("FIELD_CONFLICT", "字段冲突", StatusWarning),
    UPDATE_AFTER_DELETE("UPDATE_AFTER_DELETE", "删除后更新", StatusDanger),
    DELETE_AFTER_UPDATE("DELETE_AFTER_UPDATE", "更新后删除", StatusDanger),
    DUPLICATE_CREATE("DUPLICATE_CREATE", "重复创建", StatusWarning),
    PERMISSION_CONFLICT("PERMISSION_CONFLICT", "权限冲突", StatusDanger),
    FILE_META_CONFLICT("FILE_META_CONFLICT", "文件元数据冲突", StatusWarning),
    ACCEPTANCE_LOCKED_CONFLICT("ACCEPTANCE_LOCKED_CONFLICT", "验收锁定冲突", StatusDanger);

    companion object {
        fun fromCode(code: String): ConflictType =
            entries.find { it.code == code } ?: VERSION_CONFLICT

        fun fromCodeOrNull(code: String?): ConflictType? =
            code?.let { entries.find { e -> e.code == it } }
    }
}
